libraries/chain/backing_store/db_context_rocksdb.cpp

```diff
@@ -338,7 +338,7 @@ namespace eosio { namespace chain { namespace backing_store {
          if (past_end(session_iter)) {
             --session_iter;
             // either way, the iterator after our known table should have a primary key in this table as the previous iterator
-            if (!past_end(session_iter) || session_iter == current_session.begin()) {
+            if (past_end(session_iter)) {
```

Comment:
It seems like the original intent was off by one line (the `--session_iter` should've come after line 341).  Does this change semantically mean the same as that?

It seems like iterating past begin() would be "undefined" behavior.  The prefix may or may not match (just depends on what is there in memory), but maybe iterating past begin will give you `end`.  I couldn't fully determine that that is the case though.

Reply:
The session::iterator has logic to set the iterator to current_session.end() (@TimothyBanks chime in here), so I was letting the second past_end call handle it (since it checks for current_session.end()).  But since you bring it up, we may not want to rely on that behavior. But if we don't want to, then I think we should assert for --current_session.begin(), or at the very least assure across the board that we are doing that and document that the begin() iterating to end() logic shouldn't be expected.

Reply:
Iterators on session are circular.  So we can iterator past begin and wrap around to the end iterator.

Reply:
Gotcha, I didn't remember at which layer we were doing the wrapping (I was thinking it was being handled more towards the WASM API).